### DISCUSSION TITLE
On the development of Pirate, Modified, and Marauder ships

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#8108] [Balance, Mechanics, and Interactions]
This was brought up in PR #8104 and I posted these initial questions there, but I think it should actually be a stand alone discussion rather than something specifically on that PR.

> Alright, while I'm enjoying the artwork for all the pirate-ified ships, I feel like I need to take a step back and ask a few questions.
> 
> 1. We appear to be settling into a course where we end up with a piratified version of most human ships. Definitely all the interesting, common, and popular ones. While I enjoy the artwork, and from a lore perspective I think it does make sense that we'd have scrapyard, chopshop, and illegally souped up versions of almost every ship in human space; that is effectively aiming at doubling the number of ships in human space. Maybe more, since we've got a range of marauder ships that do several different types of modifications to a ship. This seems at odds with the "no more human ships without a very clear reason for having them."
> 2. What's the purpose of this ship, from a meta perspective? Looking at the stats, this is quite literally the gunship with slightly more hull and mass, and slightly less drag and cargo. And the mass change actually is not actually a mass change since the extra mass balances out almost exactly against the reduced outfit and cargo space. These two ships are so similar as to practically not be worth having a separate ship for it. This is almost exactly why we have variants. And I'll note that variants can have different sprites, too.
> 3. If we're going to be doing piratified versions of a lot of ships, is this the best way to go about doing it? Are there alternatives we could consider instead?
> 
> _Originally posted by @Zitchas in https://github.com/endless-sky/endless-sky/issues/8104#issuecomment-1380577408_

**Comment by Zitchas** [comment]
> 1. No, we're not piratifing everything. We have a small list of what we'll do as approved by Derpy:
> Northern: Behemoth (Mammoth, done), Aerie(Valkyrie, todo:Saugia), Gunboat(Cutlass, this).
> Core: Protector
> South: Clipper, Bastion, Blackbird.
> 
> 2. I imagine this would be somewhat a lighter, faster Firebird in role. Great firepower and fast while also quite tough (because Navy ships) so it have higher tolerance for bad day where you got jumped on by a few patrols at once.
> 
> 3. See 1.
> 
> _Originally posted by @1010todd in https://github.com/endless-sky/endless-sky/issues/8104#issuecomment-1380587624_

1. Alright, so my initial guess was accurate: Most of the popular ships, with the amusing addition of the so-unpopular-it-is-iconic Protector.

2. So the general idea of these is to give the player a bad day? That's the net impact I'm seeing. Although admitedly these would have the inverse effect in a Pirate campaign where these are the typical shipyards the player is purchasing from.

3. Uh... 1 doesn't mentione a thing about the way about accomplishing this.

-----------

To expand a bit, I was thinking "Is creating a stand alone new ship for each piratified ship actually the best way to go about it?", not "has anyone thought about which ships to piratify?" 

So, has anyone thought about alternate ways of going about this? And yes, these ideas may require new coding to get the full effect.

Some initial thoughts I've had:

A. Players like customizing ships. Instead of having "piratified gunboat" that is, quite literally, just a regular gunboat with some extra hull and gunports; why don't we have something that ties into the outfitting system. For example:
Pirate upgrades are mutually exclusive, you can only have one of them. Getting any of the pirate upgrades applies an overlay to your ship's sprite that gives it that pirate look.
- Pirate Combat Upgrade: This one adds a thin layer of hull plating to your ship, and doubles the number of available gunports. (all new gunports would share the same location as the existing gunports)
- Pirate Smuggler's Special: These mechanics have figured out how to lace your hull with strips of interference plating, in addition to adding a variety of discrete and well shielded compartments throughout your ship. +10% cargo space and the equivalent of 5 interference plating outfits.
- Pirate's Ratrod: Some people just like breaking the safety limits by installing more engines and pushing the safety limits to the max.  This reduces the ship's drag by 10%, and increases its engine capacity by 10%.
(I can see *lots* of potential upgrades along these lines.)

B. Including the list of proposed ships, that will bring us up to 18, I think, ships that are quite literally "regular ships that have been modified in ways that we don't allow the player to do. At what point do we just go "OK, let's just let the player make these modifications themselves?" Most of these changes boil down to:
- Adding more hull, shields, cargo, outfits, or types of space (all of these we have the mechanics to do already)
- Changing other ship attributes (mass and drag mostly; which we also already have mechanics to do as outfits)
- Adding more hardpoints. There's an open request for one way to handle this https://github.com/endless-sky/endless-sky/issues/4700

C. Furthering the trend of player-triggered upgrades, once we have the take ship PR merged, we can use missions to do this too: https://github.com/endless-sky/endless-sky/pull/6484 . There's a variety of ways we can use this. Ranging from "buy an outfit, and the presence of this outfit triggers an invisible mission that replaces the ship its on with the appropriate piratified variant" to "have a mechanic station with jobs that can be accepted that take the current ship and replace it with the appropriate variant of the ship.

----

On the lore side, I think all this just raises a single, important, question of lore: If the pirates can add hull, shield, gunports, reduce drag, expand engine space, and countless other changes to a ship.... Why can't anyone else in human space? I'd expect at least the folks on Rust or maybe Tarazed would be thrilled to provide some custom modifications to ships. Not to mention any one of the likely thousands of self-taught enthusiasts across the dirt belt, Rim, or South planets that are none too fond of government regulations to begin with.

**Comment by Zitchas** [comment]
Likewise, I think we really should be taking a hard look at the overall purpose of these modifications. They are being made by and for pirates, right? What is the purpose of pirates? Their goals? Generally speaking, we view them as thieves. People whose goal is to steal the cargo from merchants. Yes, some will probably want to steal the whole ship, not just the cargo. But most (especially the smaller ones) should be mainly after cargo. 

However, there's a trend, started by some of the bigger marauder ships, wherein the pirate ships are better *warships* than the warships they are based on. I think this is very problematic, because it suggests several things:
1. That pirates make better warships than the regular shipyards that make those warships.
2. That pirates are more interested in building ships that can take on warships than they are on, well, being pirates.
3. It also suggests that pirates are more interested in taking part in the capture ladder than they are on making a quick profit via stolen cargoes.

This results in a few questions;
1. Why are we portraying anarchists, child-slavers, general criminals, and smugglers as being the most advanced shipwrights in human space? Not saying that this is wrong or bad, just saying that this is what we're portraying, and it says something very significant about the motivations, capabilities, and value that mainstream human society has in regards to engineering, science, and ship building. This fits, to a large extent, with the critical portrayal of humanity as being profit driven, caring primarily about luxuries, money, and taxes (or rather, lack thereof).
2. If the pirates are, in fact, more interested in building warships to take on warships rather than better pirating ships to loot with; that suggests that they (or their backers) have larger plans. Perhaps Alphas (which already have a history of using pirates), are carefully cultivating this as way to create a widely distributed and largely independent space navy for use in future conflicts. Perhaps the Syndicate has been doing so as a longer-term version of what they did to start the civil war. Slower paced, but much lower risk. Also potential training grounds for mercenaries and security specialists they can hire as needed.
3. There's a general opinion that the capture ladder is an OP way of getting into better ships. We've got the boarding rework aiming to fix some of that, and lots, and lots of commentary ranging from "capturing ships should be impossible" through to "it should be possible but very hard to do" and "Capturing ships is fine, it should just be impossible to get *better* ships that way." As such, should we be building NPCs and NPC ships that look like they're designed and built specifically for people climbing the ladder?

This suggests a lot of things:

a) Pirates should put more emphasis on picking up cargo when their target has dumped their cargo load. When their cargo hold is full, they should **leave**. 
b) Pirate ship variants should be putting more emphasis on cargo. Unless we're going for some kind of larger "there's actually a power and a plan behind the pirates" story, we should be pulling back on the "pirates building better combat warships" angle, and putting more emphasis on "pirates building better ships to loot with." They *should* be among the best, if not *the* best, place to go for cargo-combat ships. As in, cargo ships modified to be able to intimidate other cargo ships into dumping their cargo, scooping it up, and running before the Navy can catch them. On the flip side, they should be making warships that can handle more cargo than the average warship. Rather like the Syndicate's trade-fleet escorts that are expected to help carry cargo too. For players that want to run commodities without stocking up on escorts? Pirate ships *should* be the goto for well-armed cargo runners.
c) Pirate ships should probably be less willing to engage targets that don't carry much cargo, more willing to run away from ships that out-gun them, and we should almost certainly be trying to move *away* from pirate ships outfitted as "warship killers." After all, their entire reason to exist is to get that cargo from merchants, right? Even firefly's ravagers preferred to board ships rather than blow them up. And yet our current pirates have a preponderance of heavy weaponry, missiles, and other armaments that have a *strong* tendency to blow stuff up and are generally a very poor pick for people that actually want to board their target.

**Comment by Zireael07** [comment]
> Why are we portraying anarchists, child-slavers, general criminals, and smugglers as being the most advanced shipwrights in human space?

I suspect this is because A LOT of space universes potray smugglers/criminals as modifying their ships to be better (granted, they're usually junk heaps too, but they're MODDED junk heaps LOL )

**Reply by Zitchas** [reply]
That's fair. And does make a degree of sense... I think it's the base frame that's the problem.

Taking junk heaps and souping them up to be actually decent performing ships? Great! Taking a terrible-in-combat ship and modifying it so that it's actually a passable combat ship; or at least good enough to easily win against similar ships to it's base frame? (in our case, say, having a marauder hauler that can easily disable and loot other freighters...) Those also make perfect sense. 

The more advanced the base ship is, though, the less this makes sense... Unless we're using this as a plot point to demonstrate certain things: 
- how the Republic's regulations and bureaucracy are stifling human innovation
- how humanity primarily values luxury, cost, and profit margins and thus aren't building ships to their capabilities, but instead to the lowest "ease of manufacture" and thus the highest profit margin.
- how humanity has devalued innovation in ship performance in favor of innovations in cost of manufacture. (in other words, humanity isn't stagnating: We're just getting better at innovating cheaper ways to build things instead of ways to make the things better. That's a perfectly valid (and highly useful) thing to specialize in. It just isn't very satisfying to players that have a strong incentive to have the toughest and deadliest flagship possible.)

**Comment by vitalchip** [comment]
Trying to make sense of ships is an exercise in futility, none of them were made to fulfill a role, other than being Jack of all trades so the player can use them.
